### PR TITLE
unit test: iot_ble_data_transfer.h API (cmock + unity)

### PR DESCRIPTION
**Pull request by dachalco**
unit test: iot_ble_data_transfer.h (cmock + unity)

Description
-----------

unit test for iot_ble_data_transfer.h API
100% line coverage
100% branch coverage
Related to this pull request: https://github.com/aws/amazon-freertos/pull/1720

*Also include a slight update in iot_ble_data_transfer.c to avoid NULL deref that is exercised by the test code

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/c_sdk/standard/ble/utest/iot_ble_data_transfer_utest.c [hunk 1]
```diff
@@ -0,0 +1,1495 @@
...
+{
+    bool ret = false;
+
+    ret = IotBleDataTransfer_Init();
+    TEST_ASSERT_TRUE_MESSAGE( ret, "Failed to initialize data" );
+}
+
+/*
+ * Force various failures of callback registration within _registerCallbacks
+ */
```
Comment (alfred2g) [inline libraries/c_sdk/standard/ble/utest/iot_ble_data_transfer_utest.c]: Can you give more details for the comments
possibly doxygen style?

File: libraries/c_sdk/standard/ble/utest/iot_ble_data_transfer_utest.c [hunk 2]
```diff
@@ -0,0 +1,1496 @@
+#include <stdbool.h>
+#include <pthread.h>
```
[the remark this answers is not in the document]
Comment (dachalco) [reply libraries/c_sdk/standard/ble/utest/iot_ble_data_transfer_utest.c]: thanks you! It was addressed in a later commit

File: libraries/c_sdk/standard/ble/utest/iot_ble_data_transfer_utest.c (hunk 1 above)
Comment (dachalco) [reply libraries/c_sdk/standard/ble/utest/iot_ble_data_transfer_utest.c]: Addressed in later commit

**Comment by dachalco** [comment]
Evidently my IDEs 'rebase' doesn't work as I expected, as indicated by the numerous history rewrites/merges above...but ye old command line, never fails me!

Also note how the Windows tests all pass now despite no code changes from me. i.e. either I was collateraled by some tangent issue, or there was a rebase issue.  CBMC failed because it couldn't start itself. Build check failed because of a permissions issue one the machine that tested it.

Going to close this and reopen in a different PR to reduce the commits so it's easier for review.

**Pull request closed without merging**